src/db/root.rs

```diff
@@ -1,9 +1,14 @@
 use crate::dag;
+use crate::util::rlog;
 
-pub async fn get_root(store: &dag::Store, head_name: &str) -> Result<String, GetRootError> {
+pub async fn get_root(
+    store: &dag::Store,
+    head_name: &str,
+    logger: rlog::Logger,
+) -> Result<String, GetRootError> {
     use GetRootError::*;
 
-    let read = store.read().await.map_err(ReadError)?;
+    let read = store.read(logger).await.map_err(ReadError)?;
```

Comment:
Seems like the fact that `read` takes logger by ownership is going to require callers to either clone or not have access to logger after call. Can we pass by ref?